core/server/models/user.js

```diff
@@ -567,50 +567,62 @@ User = ghostBookshelf.Model.extend({
     },
 
     // Finds the user by email, and checks the password
+    // @TODO: shorten this function and rename...
     check: function check(object) {
         var self = this,
             s;
+
         return this.getByEmail(object.email).then(function then(user) {
             if (!user) {
                 return Promise.reject(new errors.NotFoundError({message: i18n.t('errors.models.user.noUserWithEnteredEmailAddr')}));
             }
 
             if (user.get('status') !== 'locked') {
-                return bcryptCompare(object.password, user.get('password')).then(function then(matched) {
-                    if (!matched) {
-                        return Promise.resolve(self.setWarning(user, {validate: false})).then(function then(remaining) {
-                            if (remaining === 0) {
-                                // If remaining attempts = 0, the account has been locked, so show a locked account message
-                                return Promise.reject(new errors.NoPermissionError({message: i18n.t('errors.models.user.accountLocked')}));
-                            }
-
-                            s = (remaining > 1) ? 's' : '';
-                            return Promise.reject(new errors.UnauthorizedError({message: i18n.t('errors.models.user.incorrectPasswordAttempts', {remaining: remaining, s: s})}));
-
-                            // Use comma structure, not .catch, because we don't want to catch incorrect passwords
-                        }, function handleError(err) {
-                            return Promise.reject(new errors.UnauthorizedError({
-                                err: err,
-                                context: i18n.t('errors.models.user.incorrectPassword'),
-                                help: i18n.t('errors.models.user.userUpdateError.help')
-                            }));
-                        });
-                    }
-
-                    return Promise.resolve(user.set({status: 'active', last_login: new Date()}).save({validate: false}))
-                        .catch(function handleError(err) {
-                            // If we get a validation or other error during this save, catch it and log it, but don't
-                            // cause a login error because of it. The user validation is not important here.
-                            logging.error(new errors.GhostError({
-                                err: err,
-                                context: i18n.t('errors.models.user.userUpdateError.context'),
-                                help: i18n.t('errors.models.user.userUpdateError.help')
-                            }));
-
-                            return user;
-                        });
-                });
+                return self.isPasswordCorrect({plainPassword: object.password, hashedPassword: user.get('password')})
```

Comment:
no logic has changed here, we just use `isPasswordCorrect` fn and react on success or error.